lib/app/features/escape_manual/data/model/escape_manual_task.dart

```diff
@@ -1,9 +1,23 @@
 import 'package:equatable/equatable.dart';
 
+import '../../domain/entity/contact.dart';
 import 'escape_manual_task_type.dart';
 
 abstract class EscapeManualTaskModel extends Equatable {
-  const EscapeManualTaskModel();
+  /// Creates a [EscapeManualTaskModel] with the given [type] and [value].
+  /// The constructor is just to assert the type of [value] based on the [type].
+  /// coverage:ignore-start
+  const EscapeManualTaskModel({
+    required EscapeManualTaskType type,
+    dynamic value,
+  })  : assert(
+          type == EscapeManualTaskType.contacts
+              ? value is List<ContactEntity>?
+              : true,
+          'Invalid value: $value for type $type',
+        ),
+        assert(type != EscapeManualTaskType.unknown);
+  // coverage:ignore-end
```

Comment:
Pela descrição não seria algo como:

```suggestion
  const EscapeManualTaskModel({
    required EscapeManualTaskType type,
    dynamic value,
  })  : assert(
          type == EscapeManualTaskType.contacts
              ? value is List<ContactEntity>?
              : value is type,
          'Invalid value: $value for type $type',
        ),
        assert(type != EscapeManualTaskType.unknown);
  // coverage:ignore-end
```

Reply:
O  `type` é um `enum` e dependendo do valor vindo em `type` o tipo do `value` é diferente, nesse caso em questão caso o enum `type` for `EscapeManualTaskType.contacts` obrigatoriamente o tipo de `value` precisa ser `List<ContactEntity>` ou null, caso seja outro `type` ele desconsidera.

A lógica está correta, porém revendo aqui, parece meio confuso mesmo, vou ajustar a condicional e mudar a mensagem.